lib/active_merchant/billing/gateways/paymentez.rb

```diff
@@ -292,9 +293,11 @@ def headers
       end
 
       def success_from(response)
-        return false if response.include?('error')
+        return false if response.include?('error') || FAILURE_STATUS.include?(response['status'] || response.dig('transaction', 'current_status'))
```

Comment:
Make sure that  response.dig('transaction', 'current_status') is CANCELLED, REJECTED and EXPIRED would be a failed status. And current_status should be before status.

Reply:
But maybe it's better just to get rid of response['status']

Reply:
Well to clean this up should we just check that 'current_status' is APPROVED or PENDING and if not we return false?

Reply:
According to my understanding in documentation, when 'current_status' is CANCELLED and 'status_detail' is 7 the transaction is successful in the context of refund so I could not reject every CANCELLED transaction, please let me know if my understanding is wrong

Reply:
Oh okay, then I think we should pass action to this method. If the action is a refund then we should check for what you said if it's not then we should just check for APPROVED or PENDING.

Reply:
Thanks a lot for the help! Corrected